src/zod-state.ts

```diff
@@ -2,6 +2,7 @@ import { z } from "zod"
 import { CUSTOM_SCOPE_KEY, DEFAULT_SCOPE_OPTIONS, DEFAULT_TYPE_OPTIONS, FOOTER_OPTION_VALUES, Z_FOOTER_OPTIONS } from "./utils"
 
 export const Config = z.object({
+   shell: z.string().nullable().optional().default(null),
```

Comment:
I would move this to the bottom (IIRC the order effects `better-commits-init`) and change it to

`shell: z.string().default('')` 

or something like 

`overrides: z.object({ shell: z.string().optional() }).default({})`

`string().nullable().optional()` is like saying `string | null | undefined`. Props in zod-state.ts don't really use null since they're auto populated by defaults if missing, and optional is only really used inside of objects